File: python solution/neonnumber.py
def checkneon(num):
    sqr = num*num
    sum = 0
    while( sqr > 0 ):
        sum = sum + sqr%10
        sqr = sqr // 10

    if( sum == num ):
        print(f'{num} is a neon number')
    else:
        print(f'{num} is not a neon number')


num = int(input("Enter the number: "))
checkneon(num)

File: java solution/neonnumber.java
import java.util.*;

class Neonnumber{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number: ");
        int num = sc.nextInt();

        int sqr = num*num;
        int sum = 0;

        while(sqr > 0){
            sum += sqr % 10;
            sqr /= 10;
        }

        if(sum == num){
            System.out.println(num + " is a neon number");
        }else{
            System.out.println(num + " is not a neon number");
        }
    }
}